packages/runtime-core/src/components/Teleport.ts

```diff
@@ -261,7 +261,7 @@ function moveTeleport(
         move(
           (children as VNode[])[i],
           container,
-          parentAnchor,
+          parentAnchor || targetAnchor,
```

Comment:
I think it is better with put the `targetAnchor` #L88 instead of `null` .
https://github.com/vuejs/vue-next/blob/0a6105f8ce4fb6f91947fb9764aae8eec1aded10/packages/runtime-core/src/components/Teleport.ts#L185-L191